src/components/InfoBar/index.tsx

```diff
@@ -122,44 +133,13 @@ export default function InfoBar(props: Props) {
     channels,
   } = useAppSelector((state) => state.node);
   const selectedSafeAddress = useAppSelector((store) => store.safe.selectedSafeAddress);
-  const account = useAppSelector((selector) => selector.web3.account) as `0x${string}`;
   const web3Connected = useAppSelector((selector) => selector.web3.status.connected);
   const nodeConnected = useAppSelector((store) => store.auth.status.connected);
-  const wxhoprSmartContractAddress = '0xD4fdec44DB9D44B8f2b6d529620f9C0C7066A2c1';
-  const xhoprSmartContractAddress = '0xD057604A14982FE8D88c5fC25Aac3267eA142a08';
-
-  const { data: xDAI_balance } = useBalance({
-    address: account,
-    watch: true,
-  });
-  const { data: wxHOPR_balance } = useBalance({
-    address: account,
-    token: wxhoprSmartContractAddress,
-    watch: true,
-  });
-  const { data: xHOPR_balance } = useBalance({
-    address: account,
-    token: xhoprSmartContractAddress,
-    watch: true,
-  });
-
-  const safeXdaiBalance = useBalance({
-    address: selectedSafeAddress as `0x${string}`,
-    watch: true,
-  }).data?.formatted;
-  const safeWxHoprBalance = useBalance({
-    address: selectedSafeAddress as `0x${string}`,
-    token: wxhoprSmartContractAddress,
-    watch: true,
-  }).data?.formatted;
-  const safexHoprBalance = useBalance({
-    address: selectedSafeAddress as `0x${string}`,
-    token: xhoprSmartContractAddress,
-    watch: true,
-  }).data?.formatted;
+  const walletBalance = useAppSelector((store) => store.web3.balance);
+  const safeBalance = useAppSelector((store) => store.safe.balance);
```

Comment:
If one of the values in `safeBalance` and `walletBalance` is updated (e.g: xDai is updated from the safeBalance but not the other values) will it re-render the whole component? Should this be separated into different const values to avoid re-rendering if only one value is updated?

Reply:
True, it will get rerendered again, but its small enough to leave it for now. We would need to close each part of the 6 balances into seperate components to avoid that